gensim/models/word2vec.py

```diff
@@ -1251,7 +1251,10 @@ def _minimize_model(self, save_syn1 = False, save_syn1neg = False, save_syn0_loc
             del self.syn1neg
         if hasattr(self, 'syn0_lockf') and not save_syn0_lockf:
             del self.syn0_lockf
-        self.model_trimmed_post_training = True
+        if not save_syn1 or not save_syn1neg or not save_syn0_lockf:
+            self.model_trimmed_post_training = True
+        import warnings
+        warnings.warn("This method would be deprecated in the future. Use KeyedVectors to retain the previously trained word vectors instead.")
```

Comment:
Please move to the top of the function. Better text is "This method would be deprecated in the future. Keep `just_word_vectors = model.wv`  to retain just the KeyedVectors instance for read-only querying of word vectors."